949 error publishing certain contributions

minump opened the pull request:
## Description
The PR fixes error when PUT is performed on a talent with null fields. The yaml file is changed to include nullable fields for talent.

**Fixes #949**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

sandeep-ps commented on CHANGELOG.md:
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Display of Talent Self Certification fields when empty. [#913](https://github.com/rokwire/rokwire-building-blocks-api/issues/913)
 - ROKWIRE_GROUP_API_KEY changed to INTERNAL_API_KEY. [#940](https://github.com/rokwire/rokwire-building-blocks-api/issues/940)
 - Updated SECURITY.md [#946](https://github.com/rokwire/rokwire-building-blocks-api/issues/946)
+- Changed talent fields to nullable [#949](https://github.com/rokwire/rokwire-building-blocks-api/issues/949)
```
Similarly here. Please move this to the Unreleased section. Thanks.

minump replied:
updated

sandeep-ps commented:
There is still an HTTP 400 error when reviewing the Contribution (step 2,3 of steps to reproduce in the issue description). Please check. Below is the example POST Contribution request:

```
curl --location --request POST 'http://localhost:5000/contributions' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer <GitHub Access Token>' \
--data-raw '{
    "capabilities": [
        {
            "apiBaseUrl": "http://mario.name",
            "apiDocUrl": "http://alisa.info",
            "dataDeletionEndpointDetails": {
                "apiKey": "siANEfDQjPs68um",
                "deletionEndpoint": "https://jeffry.info"
            },
            "deploymentDetails": {
                "authMethod": "OIDC",
                "databaseDetails": "MongoDB",
                "dockerImageName": "asperiores",
                "environmentVariables": [
                    {
                        "key": "maxime",
                        "value": "repellendus"
                    },
                    {
                        "key": "repellendus",
                        "value": "rerum"
                    }
                ],
                "location": "internal"
            },
            "description": "Dolore dolores doloribus aut error quo dolorem suscipit molestias eum.",
            "healthCheckUrl": "https://gennaro.org",
            "isOpenSource": true,
            "sourceRepoUrl": "https://sincere.info",
            "name": "Sleek Wooden Cheese 492",
            "versionUrl": "https://carmelo.biz",
            "version": "2.7.9",
            "id": "4e4c768d-17f6-4703-a730-76ebc0f1cd99"
        }
    ],
    "contacts": [
        {
            "email": "Zackary_Terry@example.net",
            "name": "Opal Herman",
            "officialAddress": "9683 Spencer Turnpike",
            "organization": "Wiza, Hackett and Harris",
            "phone": "652-816-3137"
        }
    ],
    "contributionAdmins": [
        "sandeep-ps"
    ],
    "contributors": [
        {
            "affiliation": {
                "address": "1885 Mueller Common",
                "contributorType": "Organization",
                "email": "Coby77@example.org",
                "name": "Ullrich - Murray",
                "phone": "946-858-4256"
            },
            "contributorType": "Person",
            "email": "Demetris.Wehner23@example.org",
            "firstName": "Montana",
            "lastName": "Maggio",
            "middleName": "Rocky",
            "phone": "326-709-0457"
        },
        {
            "address": "236 Myrtis Road",
            "contributorType": "Organization",
            "email": "Adalberto_Glover49@example.org",
            "name": "Bayer, Wuckert and Reilly",
            "phone": "856-392-6638"
        }
    ],
    "longDescription": "Molestiae sequi nobis est iure et sunt laboriosam sapiente. Eum ut omnis aut vitae in ipsa nihil. Maiores dignissimos omnis non provident maxime aliquid voluptatem. Magni nostrum quisquam. Possimus ducimus laudantium velit amet tempore tempore alias.",
    "status": "Submitted",
    "talents": [
        {
            "dataDescription": "Quo omnis consequatur soluta esse.",
            "endDate": "Thu May 18 2023 19:58:35 GMT-0500 (Central Daylight Time)",
            "id": "715114b9-1710-4754-84cb-5b27cbcaa57d",
            "longDescription": "Rerum corporis laborum consequatur in sapiente necessitatibus quis quo. Sit quos nesciunt voluptates et. Ducimus magni delectus et minus illo natus in laboriosam. Aut voluptatem consectetur qui repellat quae nisi commodi rerum voluptas.",
            "minEndUserRoles": [
                "visitor"
            ],
            "minUserPrivacyLevel": 4,
            "name": "Generic Rubber Mouse 165",
            "requiredBuildingBlocks": [
                "Sports"
            ],
            "selfCertification": {
                "dataDeletionUponRequest": "Omnis eius et rerum.",
                "discloseAds": "Eum et hic unde ea veniam.",
                "discloseImageRights": "Atque dolorem porro et est autem laborum sit inventore reiciendis.",
                "discloseSponsors": "Ipsum quae laudantium explicabo distinctio dignissimos illum quam animi molestias.",
                "respectingUserPrivacySetting": "Atque delectus qui temporibus eaque mollitia sed."
            },
            "shortDescription": "Similique laudantium eum voluptatem eos consequatur labore omnis amet debitis.",
            "startDate": "Thu Jan 19 2023 10:26:30 GMT-0600 (Central Standard Time)"
        }
    ],
    "name": "Unbranded Soft Towels 897",
    "shortDescription": "Facere necessitatibus excepturi asperiores."
}'
```

This is the error:

```
validation error: None is not of type 'array' - 'talents.0.requiredCapabilities'
2022-08-16T03:46:01.558Z DEBUG   [Thread-23 ] : connexion.apis.abstract - Getting data and status code
2022-08-16T03:46:01.558Z DEBUG   [Thread-23 ] : connexion.apis.abstract - Prepared body and status code (400)
```

minump commented:
> There is still an HTTP 400 error when reviewing the Contribution (step 2,3 of steps to reproduce in the issue description). Please check. Below is the example POST Contribution request:
> 
> ```
> curl --location --request POST 'http://localhost:5000/contributions' \
> --header 'Content-Type: application/json' \
> --header 'Authorization: Bearer <GitHub Access Token>' \
> --data-raw '{
>     "capabilities": [
>         {
>             "apiBaseUrl": "http://mario.name",
>             "apiDocUrl": "http://alisa.info",
>             "dataDeletionEndpointDetails": {
>                 "apiKey": "siANEfDQjPs68um",
>                 "deletionEndpoint": "https://jeffry.info"
>             },
>             "deploymentDetails": {
>                 "authMethod": "OIDC",
>                 "databaseDetails": "MongoDB",
>                 "dockerImageName": "asperiores",
>                 "environmentVariables": [
>                     {
>                         "key": "maxime",
>                         "value": "repellendus"
>                     },
>                     {
>                         "key": "repellendus",
>                         "value": "rerum"
>                     }
>                 ],
>                 "location": "internal"
>             },
>             "description": "Dolore dolores doloribus aut error quo dolorem suscipit molestias eum.",
>             "healthCheckUrl": "https://gennaro.org",
>             "isOpenSource": true,
>             "sourceRepoUrl": "https://sincere.info",
>             "name": "Sleek Wooden Cheese 492",
>             "versionUrl": "https://carmelo.biz",
>             "version": "2.7.9",
>             "id": "4e4c768d-17f6-4703-a730-76ebc0f1cd99"
>         }
>     ],
>     "contacts": [
>         {
>             "email": "Zackary_Terry@example.net",
>             "name": "Opal Herman",
>             "officialAddress": "9683 Spencer Turnpike",
>             "organization": "Wiza, Hackett and Harris",
>             "phone": "652-816-3137"
>         }
>     ],
>     "contributionAdmins": [
>         "sandeep-ps"
>     ],
>     "contributors": [
>         {
>             "affiliation": {
>                 "address": "1885 Mueller Common",
>                 "contributorType": "Organization",
>                 "email": "Coby77@example.org",
>                 "name": "Ullrich - Murray",
>                 "phone": "946-858-4256"
>             },
>             "contributorType": "Person",
>             "email": "Demetris.Wehner23@example.org",
>             "firstName": "Montana",
>             "lastName": "Maggio",
>             "middleName": "Rocky",
>             "phone": "326-709-0457"
>         },
>         {
>             "address": "236 Myrtis Road",
>             "contributorType": "Organization",
>             "email": "Adalberto_Glover49@example.org",
>             "name": "Bayer, Wuckert and Reilly",
>             "phone": "856-392-6638"
>         }
>     ],
>     "longDescription": "Molestiae sequi nobis est iure et sunt laboriosam sapiente. Eum ut omnis aut vitae in ipsa nihil. Maiores dignissimos omnis non provident maxime aliquid voluptatem. Magni nostrum quisquam. Possimus ducimus laudantium velit amet tempore tempore alias.",
>     "status": "Submitted",
>     "talents": [
>         {
>             "dataDescription": "Quo omnis consequatur soluta esse.",
>             "endDate": "Thu May 18 2023 19:58:35 GMT-0500 (Central Daylight Time)",
>             "id": "715114b9-1710-4754-84cb-5b27cbcaa57d",
>             "longDescription": "Rerum corporis laborum consequatur in sapiente necessitatibus quis quo. Sit quos nesciunt voluptates et. Ducimus magni delectus et minus illo natus in laboriosam. Aut voluptatem consectetur qui repellat quae nisi commodi rerum voluptas.",
>             "minEndUserRoles": [
>                 "visitor"
>             ],
>             "minUserPrivacyLevel": 4,
>             "name": "Generic Rubber Mouse 165",
>             "requiredBuildingBlocks": [
>                 "Sports"
>             ],
>             "selfCertification": {
>                 "dataDeletionUponRequest": "Omnis eius et rerum.",
>                 "discloseAds": "Eum et hic unde ea veniam.",
>                 "discloseImageRights": "Atque dolorem porro et est autem laborum sit inventore reiciendis.",
>                 "discloseSponsors": "Ipsum quae laudantium explicabo distinctio dignissimos illum quam animi molestias.",
>                 "respectingUserPrivacySetting": "Atque delectus qui temporibus eaque mollitia sed."
>             },
>             "shortDescription": "Similique laudantium eum voluptatem eos consequatur labore omnis amet debitis.",
>             "startDate": "Thu Jan 19 2023 10:26:30 GMT-0600 (Central Standard Time)"
>         }
>     ],
>     "name": "Unbranded Soft Towels 897",
>     "shortDescription": "Facere necessitatibus excepturi asperiores."
> }'
> ```
> 
> This is the error:
> 
> ```
> validation error: None is not of type 'array' - 'talents.0.requiredCapabilities'
> 2022-08-16T03:46:01.558Z DEBUG   [Thread-23 ] : connexion.apis.abstract - Getting data and status code
> 2022-08-16T03:46:01.558Z DEBUG   [Thread-23 ] : connexion.apis.abstract - Prepared body and status code (400)
> ```

changed the default value to [] (empty list) in yaml file. Works now.
Updated.

minump commented:
Tested again. nullable : true gives error 400 (for Sandeep), but default : [] also gives error 400 (for Minu). 
Need further testing.

minump commented:
converting to draft for further testing

minump commented:
This curl POST command worked for me : 
curl --location --request POST 'http://localhost:5000/contributions' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer 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' \
--data-raw '{
    "capabilities": [
        {
            "apiBaseUrl": "http://mario.name",
            "apiDocUrl": "http://alisa.info",
            "dataDeletionEndpointDetails": {
                "apiKey": "siANEfDQjPs68um",
                "deletionEndpoint": "https://jeffry.info"
            },
            "deploymentDetails": {
                "authMethod": "OIDC",
                "databaseDetails": "MongoDB",
                "dockerImageName": "asperiores",
                "environmentVariables": [
                    {
                        "key": "maxime",
                        "value": "repellendus"
                    },
                    {
                        "key": "repellendus",
                        "value": "rerum"
                    }
                ],
                "location": "internal"
            },
            "description": "Dolore dolores doloribus aut error quo dolorem suscipit molestias eum.",
            "healthCheckUrl": "https://gennaro.org",
            "isOpenSource": true,
            "sourceRepoUrl": "https://sincere.info",
            "name": "Sleek Wooden Cheese 492",
            "versionUrl": "https://carmelo.biz",
            "version": "2.7.9",
            "id": "824e0448-69ef-41a0-bbcb-2b0a06a15233"
        }
    ],
    "contacts": [
        {
            "email": "Zackary_Terry@example.net",
            "name": "Opal Herman",
            "officialAddress": "9683 Spencer Turnpike",
            "organization": "Wiza, Hackett and Harris",
            "phone": "652-816-3137"
        }
    ],
    "contributionAdmins": [
        "sandeep-ps", "minump"
    ],
    "contributors": [
        {
            "affiliation": {
                "address": "1885 Mueller Common",
                "contributorType": "Organization",
                "email": "Coby77@example.org",
                "name": "Ullrich - Murray",
                "phone": "946-858-4256"
            },
            "contributorType": "Person",
            "email": "Demetris.Wehner23@example.org",
            "firstName": "Montana",
            "lastName": "Maggio",
            "middleName": "Rocky",
            "phone": "326-709-0457"
        },
        {
            "address": "236 Myrtis Road",
            "contributorType": "Organization",
            "email": "Adalberto_Glover49@example.org",
            "name": "Bayer, Wuckert and Reilly",
            "phone": "856-392-6638"
        }
    ],
    "longDescription": "Molestiae sequi nobis est iure et sunt laboriosam sapiente. Eum ut omnis aut vitae in ipsa nihil. Maiores dignissimos omnis non provident maxime aliquid voluptatem. Magni nostrum quisquam. Possimus ducimus laudantium velit amet tempore tempore alias.",
    "status": "Submitted",
    "talents": [
        {
            "dataDescription": "Quo omnis consequatur soluta esse.",
            "endDate": "Thu May 18 2023 19:58:35 GMT-0500 (Central Daylight Time)",
            "id": "715114b9-1710-4754-84cb-5b27cbcaa57d",
            "longDescription": "Rerum corporis laborum consequatur in sapiente necessitatibus quis quo. Sit quos nesciunt voluptates et. Ducimus magni delectus et minus illo natus in laboriosam. Aut voluptatem consectetur qui repellat quae nisi commodi rerum voluptas.",
            "minEndUserRoles": [
                "visitor"
            ],
            "minUserPrivacyLevel": 4,
            "name": "Generic Rubber Mouse 165",
            "requiredBuildingBlocks": [
                "Sports"
            ],
            "selfCertification": {
                "dataDeletionUponRequest": "Omnis eius et rerum.",
                "discloseAds": "Eum et hic unde ea veniam.",
                "discloseImageRights": "Atque dolorem porro et est autem laborum sit inventore reiciendis.",
                "discloseSponsors": "Ipsum quae laudantium explicabo distinctio dignissimos illum quam animi molestias.",
                "respectingUserPrivacySetting": "Atque delectus qui temporibus eaque mollitia sed."
            },
            "shortDescription": "Similique laudantium eum voluptatem eos consequatur labore omnis amet debitis.",
            "startDate": "Thu Jan 19 2023 10:26:30 GMT-0600 (Central Standard Time)"
        }
    ],
    "name": "Unbranded Soft Towels 897",
    "shortDescription": "Facere necessitatibus excepturi asperiores."
}'

minump commented:
Converting to ready for review for further testing.

minump commented:
> This curl POST command worked for me :
> 
> ```
> curl --location --request POST 'http://localhost:5000/contributions' \
> --header 'Content-Type: application/json' \
> --header 'Authorization: Bearer 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' \
> --data-raw '{
>     "capabilities": [
>         {
>             "apiBaseUrl": "http://mario.name",
>             "apiDocUrl": "http://alisa.info",
>             "dataDeletionEndpointDetails": {
>                 "apiKey": "siANEfDQjPs68um",
>                 "deletionEndpoint": "https://jeffry.info"
>             },
>             "deploymentDetails": {
>                 "authMethod": "OIDC",
>                 "databaseDetails": "MongoDB",
>                 "dockerImageName": "asperiores",
>                 "environmentVariables": [
>                     {
>                         "key": "maxime",
>                         "value": "repellendus"
>                     },
>                     {
>                         "key": "repellendus",
>                         "value": "rerum"
>                     }
>                 ],
>                 "location": "internal"
>             },
>             "description": "Dolore dolores doloribus aut error quo dolorem suscipit molestias eum.",
>             "healthCheckUrl": "https://gennaro.org",
>             "isOpenSource": true,
>             "sourceRepoUrl": "https://sincere.info",
>             "name": "Sleek Wooden Cheese 492",
>             "versionUrl": "https://carmelo.biz",
>             "version": "2.7.9",
>             "id": "824e0448-69ef-41a0-bbcb-2b0a06a15233"
>         }
>     ],
>     "contacts": [
>         {
>             "email": "Zackary_Terry@example.net",
>             "name": "Opal Herman",
>             "officialAddress": "9683 Spencer Turnpike",
>             "organization": "Wiza, Hackett and Harris",
>             "phone": "652-816-3137"
>         }
>     ],
>     "contributionAdmins": [
>         "sandeep-ps", "minump"
>     ],
>     "contributors": [
>         {
>             "affiliation": {
>                 "address": "1885 Mueller Common",
>                 "contributorType": "Organization",
>                 "email": "Coby77@example.org",
>                 "name": "Ullrich - Murray",
>                 "phone": "946-858-4256"
>             },
>             "contributorType": "Person",
>             "email": "Demetris.Wehner23@example.org",
>             "firstName": "Montana",
>             "lastName": "Maggio",
>             "middleName": "Rocky",
>             "phone": "326-709-0457"
>         },
>         {
>             "address": "236 Myrtis Road",
>             "contributorType": "Organization",
>             "email": "Adalberto_Glover49@example.org",
>             "name": "Bayer, Wuckert and Reilly",
>             "phone": "856-392-6638"
>         }
>     ],
>     "longDescription": "Molestiae sequi nobis est iure et sunt laboriosam sapiente. Eum ut omnis aut vitae in ipsa nihil. Maiores dignissimos omnis non provident maxime aliquid voluptatem. Magni nostrum quisquam. Possimus ducimus laudantium velit amet tempore tempore alias.",
>     "status": "Submitted",
>     "talents": [
>         {
>             "dataDescription": "Quo omnis consequatur soluta esse.",
>             "endDate": "Thu May 18 2023 19:58:35 GMT-0500 (Central Daylight Time)",
>             "id": "715114b9-1710-4754-84cb-5b27cbcaa57d",
>             "longDescription": "Rerum corporis laborum consequatur in sapiente necessitatibus quis quo. Sit quos nesciunt voluptates et. Ducimus magni delectus et minus illo natus in laboriosam. Aut voluptatem consectetur qui repellat quae nisi commodi rerum voluptas.",
>             "minEndUserRoles": [
>                 "visitor"
>             ],
>             "minUserPrivacyLevel": 4,
>             "name": "Generic Rubber Mouse 165",
>             "requiredBuildingBlocks": [
>                 "Sports"
>             ],
>             "selfCertification": {
>                 "dataDeletionUponRequest": "Omnis eius et rerum.",
>                 "discloseAds": "Eum et hic unde ea veniam.",
>                 "discloseImageRights": "Atque dolorem porro et est autem laborum sit inventore reiciendis.",
>                 "discloseSponsors": "Ipsum quae laudantium explicabo distinctio dignissimos illum quam animi molestias.",
>                 "respectingUserPrivacySetting": "Atque delectus qui temporibus eaque mollitia sed."
>             },
>             "shortDescription": "Similique laudantium eum voluptatem eos consequatur labore omnis amet debitis.",
>             "startDate": "Thu Jan 19 2023 10:26:30 GMT-0600 (Central Standard Time)"
>         }
>     ],
>     "name": "Unbranded Soft Towels 897",
>     "shortDescription": "Facere necessitatibus excepturi asperiores."
> }'
> ```

The contribution dataset stored in mongodb as per the above POST is : 
```
{"capabilities": [{"apiBaseUrl": null, "apiDocUrl": "http://alisa.info", "dataDeletionEndpointDetails": {"apiKey": "siANEfDQjPs68um", "deletionEndpoint": "https://jeffry.info", "description": null}, "deploymentDetails": {"authMethod": "OIDC", "databaseDetails": "MongoDB", "dockerImageName": "asperiores", "environmentVariables": [{"key": "maxime", "value": "repellendus"}, {"key": "repellendus", "value": "rerum"}], "location": "internal"}, "description": "Dolore dolores doloribus aut error quo dolorem suscipit molestias eum.", "healthCheckUrl": "https://gennaro.org", "icon": null, "id": "0e8ca703-b6e8-4da9-a72d-ba4b10606661", "isOpenSource": true, "name": "Sleek Wooden Cheese 492", "sourceRepoUrl": "https://sincere.info", "version": "2.7.9", "versionUrl": "https://carmelo.biz"}], "contacts": [{"email": "Zackary_Terry@example.net", "name": "Opal Herman", "officialAddress": "9683 Spencer Turnpike", "organization": "Wiza, Hackett and Harris", "phone": "652-816-3137"}], "contributionAdmins": ["sandeep-ps", "minump"], "contributors": [{"affiliation": {"address": "1885 Mueller Common", "contributorType": "Organization", "email": "Coby77@example.org", "name": "Ullrich - Murray", "phone": "946-858-4256"}, "contributorType": "Person", "email": "Demetris.Wehner23@example.org", "firstName": "Montana", "lastName": "Maggio", "middleName": "Rocky", "phone": "326-709-0457"}, {"address": "236 Myrtis Road", "contributorType": "Organization", "email": "Adalberto_Glover49@example.org", "name": "Bayer, Wuckert and Reilly", "phone": "856-392-6638"}], "dateCreated": "2022/10/27T10:53:58", "dateModified": "2022/10/27T10:53:58", "longDescription": "Molestiae sequi nobis est iure et sunt laboriosam sapiente. Eum ut omnis aut vitae in ipsa nihil. Maiores dignissimos omnis non provident maxime aliquid voluptatem. Magni nostrum quisquam. Possimus ducimus laudantium velit amet tempore tempore alias.", "name": "Unbranded Soft Towels 897", "review": null, "shortDescription": "Facere necessitatibus excepturi asperiores.", "status": "Submitted", "talents": [{"dataDescription": "Quo omnis consequatur soluta esse.", "endDate": "Thu May 18 2023 19:58:35 GMT-0500 (Central Daylight Time)", "icon": null, "id": "715114b9-1710-4754-84cb-5b27cbcaa57d", "longDescription": "Rerum corporis laborum consequatur in sapiente necessitatibus quis quo. Sit quos nesciunt voluptates et. Ducimus magni delectus et minus illo natus in laboriosam. Aut voluptatem consectetur qui repellat quae nisi commodi rerum voluptas.", "minEndUserRoles": ["visitor"], "minUserPrivacyLevel": 4, "name": "Generic Rubber Mouse 165", "requiredBuildingBlocks": ["Sports"], "requiredCapabilities": null, "selfCertification": {"dataDeletionUponRequest": "Omnis eius et rerum.", "discloseAds": "Eum et hic unde ea veniam.", "discloseImageRights": "Atque dolorem porro et est autem laborum sit inventore reiciendis.", "discloseSponsors": "Ipsum quae laudantium explicabo distinctio dignissimos illum quam animi molestias.", "respectingUserPrivacySetting": "Atque delectus qui temporibus eaque mollitia sed."}, "shortDescription": "Similique laudantium eum voluptatem eos consequatur labore omnis amet debitis.", "startDate": "Thu Jan 19 2023 10:26:30 GMT-0600 (Central Standard Time)"}]}
```

talents.requiredCapabilities : null. -- This is expected as per yaml file.

sandeep-ps commented:
Tested the previously failed test case (https://github.com/rokwire/rokwire-building-blocks-api/pull/965#issuecomment-1216111681) and a few other Contribution POSTs from Postman and later approved them from Catalog as per the steps to repeat in the issue description. The bug seems to be fixed. Thanks.

sandeep-ps reviewed (approved):
Approving. Thanks.

minump commented:
Merging with develop
